packages/superset-ui-query/src/extractExtras.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable camelcase */
+import { isDruidFormData, QueryFormData } from './types/QueryFormData';
+import { QueryObject } from './types/Query';
+
+export default function extractExtras(formData: QueryFormData): Partial<QueryObject> {
+  const partialQueryObject: Partial<QueryObject> = {
+    filters: formData.filters || [],
+    extras: formData.extras || {},
+  };
+
+  const reservedColumnsToQueryField: Record<string, keyof QueryObject> = {
+    __time_range: 'time_range',
+    __time_col: 'granularity_sqla',
+    __time_grain: 'time_grain_sqla',
+    __time_origin: 'druid_time_origin',
+    __granularity: 'granularity',
+  };
+
+  (formData.extra_filters || []).forEach(filter => {
+    if (filter.col in reservedColumnsToQueryField) {
+      const queryField = reservedColumnsToQueryField[filter.col];
+      partialQueryObject[queryField] = filter.val;
+    } else {
+      // @ts-ignore
+      partialQueryObject.filters.push(filter);
+    }
+  });
+
+  // map to undeprecated names and remove deprecated fields
+  if (isDruidFormData(formData) && !partialQueryObject.druid_time_origin) {
+    partialQueryObject.extras = {
+      druid_time_origin: formData.druid_time_origin,
```

Comment:
`druid_time_origin` is optional on this data type. Should a default be specified here, e.g. `druid_time_origin: formData.druid_time_origin || someDefault,`?

Reply:
The backend expects most of these to be undefined if they haven't been given an explicit value (`time_grain`, `druid_time_origin` etc). So these should be safe to leave as is.